packages/atomic/src/components/insight/atomic-insight-result-list/atomic-insight-result-list.tsx

```diff
@@ -0,0 +1,222 @@
+ * @part result-table-row-even - The element containing the cells of an even row in the result table's body
+ * @part result-table-cell - The element representing a cell of the result table's body
+ */
+@Component({
+  tag: 'atomic-insight-result-list',
+  // TODO:
+  styleUrl: 'atomic-insight-result-list.pcss',
+  shadow: true,
+})
+export class AtomicInsightResultList {
+  @InitializeBindings() public bindings!: InsightBindings;
+  public resultList!: InsightResultList;
+  private resultTemplatesManager!: InsightResultTemplatesManager<TemplateContent>;
+  @State() public ready = false;
+  @Element() public host!: HTMLDivElement;
+  // TODO:
+  // public resultsPerPage!: ResultsPerPage;
+  // public listWrapperRef?: HTMLDivElement;
+
+  @State() public templateHasError = false;
+
+  @BindStateToController('resultList')
+  @State()
+  public resultListState!: InsightResultListState;
+
+  // TODO:
+  // @BindStateToController('resultsPerPage')
+  // @State()
+  // public resultsPerPageState!: ResultsPerPageState;
+
+  @State() public error!: Error;
+
+  // TODO:
+  // @FocusTarget() nextNewResultTarget!: FocusTargetController;
+
+  // TODO:
+  // private loadingFlag = randomID('firstResultLoaded-');
+
+  /**
+   * The spacing of various elements in the result list, including the gap between results, the gap between parts of a result, and the font sizes of different parts in a result.
+   */
+  @Prop({reflect: true}) density: ResultDisplayDensity = 'normal';
+  /**
+   * The expected size of the image displayed in the results.
+   */
+  @Prop({reflect: true}) imageSize: ResultDisplayImageSize = 'icon';
+
+  // TODO:
+  // /**
+  //  * @internal
+  //  */
+  // @Method() public async focusOnNextNewResult() {
+  //   this.resultListCommon.focusOnNextNewResult(this.resultListState);
+  // }
+
+  public async initialize() {
+    this.resultList = buildInsightResultList(this.bindings.engine, {
+      options: {
+        fieldsToInclude: this.bindings.store.state.fieldsToInclude || undefined,
+      },
+    });
+    this.registerResultTemplates();
+    // TODO:
+    // this.resultsPerPage = buildResultsPerPage(this.bindings.engine);
+    // this.bindings.store.registerResultList(this);
+  }
+
+  public getTemplate(result: InsightResult) {
+    return this.resultTemplatesManager.selectTemplate(result);
+  }
+
+  private async registerResultTemplates() {
+    this.resultTemplatesManager = insightBuildResultTemplatesManager(
+      this.bindings.engine
+    );
+    const elements = this.host.querySelectorAll(
+      'atomic-insight-result-template'
+    ) as NodeListOf<TemplateElement>;
+    const customTemplates = await Promise.all(
+      Array.from(elements).map(async (resultTemplateElement) => {
+        const template = await resultTemplateElement.getTemplate();
+        if (!template) {
+          this.templateHasError = true;
+        }
+        return template;
+      })
```

Comment:
Perhaps this could be moved to the common result list. The common result list could take the tag name as an option.